launch/lifelong_launch.py

```diff
@@ -1,17 +1,71 @@
+                            RegisterEventHandler)
+from launch.conditions import IfCondition
+from launch.events import matches_action
+from launch.substitutions import (AndSubstitution, LaunchConfiguration,
+                                  NotSubstitution)
+from launch_ros.actions import LifecycleNode
+from launch_ros.event_handlers import OnStateTransition
+from launch_ros.events.lifecycle import ChangeState
+from lifecycle_msgs.msg import Transition
 
 
 def generate_launch_description():
-    return LaunchDescription([
-        launch_ros.actions.Node(
+    autostart = LaunchConfiguration('autostart')
+    use_lifecycle_manager = LaunchConfiguration("use_lifecycle_manager")
+
+    declare_autostart_cmd = DeclareLaunchArgument(
+        'autostart', default_value='true',
+        description='Automatically startup the slamtoolbox. '
+                    'Ignored when use_lifecycle_manager is true.')
+    declare_use_lifecycle_manager = DeclareLaunchArgument(
+        'use_lifecycle_manager', default_value='false',
+        description='Enable bond connection duaring node activation')
+
+    start_lifelong_slam_toolbox_node = LifecycleNode(
           parameters=[
-            get_package_share_directory("slam_toolbox") + '/config/mapper_params_lifelong.yaml'
+            get_package_share_directory("slam_toolbox") + '/config/mapper_params_lifelong.yaml',
+            {'use_lifecycle_manager': use_lifecycle_manager}
           ],
           package='slam_toolbox',
           executable='lifelong_slam_toolbox_node',
           name='slam_toolbox',
-          output='screen'
-        )
-    ])
+          output='screen',
+          namespace=''
+    )
+
+    configure_event = EmitEvent(
+        event=ChangeState(
+            lifecycle_node_matcher=matches_action(start_lifelong_slam_toolbox_node),
+            transition_id=Transition.TRANSITION_CONFIGURE
+        ),
+        condition=IfCondition(AndSubstitution(autostart, NotSubstitution(use_lifecycle_manager)))
+    )
+
+    activate_event = RegisterEventHandler(
+        OnStateTransition(
+            target_lifecycle_node=start_lifelong_slam_toolbox_node,
+            start_state="configuring",
+            goal_state="inactive",
+            entities=[
+                LogInfo(msg="[LifecycleLaunch] Slamtoolbox node is activating."),
+                EmitEvent(event=ChangeState(
+                    lifecycle_node_matcher=matches_action(start_lifelong_slam_toolbox_node),
+                    transition_id=Transition.TRANSITION_ACTIVATE
+                ))
+            ]
+        ),
+        condition=IfCondition(AndSubstitution(autostart, NotSubstitution(use_lifecycle_manager)))
+    )
+
+    ld = LaunchDescription()
+
+    ld.add_action(declare_autostart_cmd)
+    ld.add_action(declare_use_lifecycle_manager)
+    ld.add_action(start_lifelong_slam_toolbox_node)
+    ld.add_action(configure_event)
+    ld.add_action(activate_event)
+
+    return ld
```

Comment:
EOF line

Reply:
Fix for all launch files

Reply:
fixed [e832151532292d39b63c16fa7e8f5a46847e77bf]